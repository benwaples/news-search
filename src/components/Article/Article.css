.Article {
  width: 350px;
  margin: 10px;
  padding: 8px;

  display: flex;
  flex-direction: column;
  justify-content: space-between;

  background-color: azure;
  border: black solid 1px;
  border-radius: 10px;


  img {
    width: 100%
  }

  img:hover {
    transform: scale(1.09);
  }

  div {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-content: center;

    h1{
      margin: 0;
    }
    p {
      margin: 0;
    }
  }

}