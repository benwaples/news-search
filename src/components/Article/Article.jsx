import React from 'react';
import PropTypes from 'prop-types';
import styles from './Article.css';

function Article({ article }) {
  return (
    <li className={styles.Article}>
      <h1>{article.title}</h1>
      <p>{article.source}</p>
      <p>{article.author}</p>
      <p>{article.content}</p>
      <a href={article.url}>Read More</a>
    </li>
  );
}

Article.propTypes = {
  article: PropTypes.objectOf({
    sourceName: PropTypes.string,
    title: PropTypes.string,
    author: PropTypes.string,
    content: PropTypes.string,
    url: PropTypes.string
  }).isRequired
};

export default Article;

