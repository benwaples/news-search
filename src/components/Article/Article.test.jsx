import React from 'react';
import { 
  cleanup, 
  fireEvent, 
  render, 
  screen, 
  waitFor 
} from '@testing-library/react';
import Article from './Article';

describe('Article component testing', () => {
  afterEach(() => cleanup());
  it('renders one article', () => {
    const article = {
      source: 'article source',
      title: 'test title',
      author: 'fake af author',
      content: 'some true stuff',
      imageUrl: 'https://fakeplaceholder.image.url.you/couldnt',
      url: 'google.com',
  
    };
    const { asFragment } = render(<Article article={article} />);

    screen.getByText('test title');

    expect(asFragment()).toMatchSnapshot();
  });
});


