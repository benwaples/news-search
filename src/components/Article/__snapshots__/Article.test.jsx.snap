// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article component testing renders one article 1`] = `
<DocumentFragment>
  <li
    class="Article"
  >
    <h1
      data-testid="article"
    >
      test title
    </h1>
    <p>
      article source
    </p>
    <p>
      fake af author
    </p>
    <p>
      some true stuff
    </p>
    <a
      href="google.com"
    >
      <img
        alt="test title"
        src="https://fakeplaceholder.image.url.you/couldnt"
      />
    </a>
    <p>
      click above to see more
    </p>
  </li>
</DocumentFragment>
`;
